packages/strapi-design-system/src/InputJSON/InputJSON.js

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { Field, FieldLabel, FieldError } from '../Field';
+import { Box } from '../Box';
+import { Stack } from '../Stack';
+import { addMarks, filterMarks, lineHighlightMark } from './decorationExtension';
+import * as jsonlint from './utils/jsonLint';
+import { JsonComponent } from './JsonComponent';
+
+const WAIT = 500;
+
+const StyledBox = styled(Box)`
+  border-radius: ${({ theme }) => theme.borderRadius};
+  outline: 1px solid ${({ theme, error }) => (error ? theme.colors.danger600 : 'transparent')};
+`;
+
+export const InputJSON = ({ id, label, value, error, theme, onChange, editable }) => {
+  const editorState = useRef(null);
+  const editorView = useRef(null);
+  const timerRef = useRef();
+
+  const getContentAtLine = (line) => {
+    return editorState.current?.doc?.line(line);
+  };
+
+  const markSelection = useCallback(({ message }) => {
+    let line = parseInt(message.split(':')[0].split('line ')[1], 10);
```

Comment:
I find this line really hard to read and understand what you're trying to achieve 🤔

Reply:
So parse error we get is of this format, and for highlighting we need to get the line number mentioned here. I know this is not readable, I will try to simplify by breaking it on separate lines. But if you have any other solution please suggest.
```
Error: Parse error on line 8:
...,      "b":6,""   }]
---------------------^
Expecting ':', got '}'
```